src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala

```diff
@@ -161,6 +162,40 @@ class ErgoTransactionSpec extends ErgoPropertyTest {
     }
   }
 
+  property("assets usage correctly affects transaction total cost") {
+    val txGen = validErgoTransactionGenTemplate(1, 1, 8, 16)
+    forAll(txGen) { case (from, tx) =>
```

Comment:
Please regroup test and write comments, what exactly are you checking here. e.g.
```
// adding new token increases transaction cost by 2 TokenAccessCostDefault (one more group + one more token in total)
val txMod0 = tx.copy(inputs = tx.inputs.init :+ txInMod0)
val inputIncTxCost0 = txMod0.statefulValidity(from.init :+ modifiedIn0, emptyStateContext).get
(inputIncTxCost0 - initTxCost) shouldEqual Parameters.TokenAccessCostDefault * 2

// adding existing token increases transaction cost by TokenAccessCost 
...

```